internal/appsec/listener/grpcsec/grpc.go

```diff
@@ -114,14 +121,16 @@ func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types
 		// UserIDOperation happens when appsec.SetUser() is called. We run the WAF and apply actions to
 		// see if the associated user should be blocked. Since we don't control the execution flow in this case
 		// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
-		dyngo.On(op, func(userIDOp *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
+		dyngo.On(op, func(op *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
 			values := map[string]any{
 				httpsec.UserIDAddr: args.UserID,
 			}
 			wafResult := shared.RunWAF(wafCtx, waf.RunAddressData{Persistent: values})
-			if wafResult.HasActions() || wafResult.HasEvents() {
-				shared.ProcessActions(userIDOp, wafResult.Actions)
-				shared.AddSecurityEvents(&op.SecurityEventsHolder, l.limiter, wafResult.Events)
+			if wafResult.HasEvents() {
+				addEvents(wafResult.Events)
+			}
+			if wafResult.HasActions() {
+				shared.ProcessActions(op, wafResult.Actions)
 				log.Debug("appsec: WAF detected an authenticated user attack: %s", args.UserID)
 			}
```

Comment:
the logs need to go up
```suggestion
			if wafResult.HasEvents() {
				addEvents(wafResult.Events)
				log.Debug("appsec: WAF detected an authenticated user attack: %s", args.UserID)
			}
			if wafResult.HasActions() {
				shared.ProcessActions(op, wafResult.Actions)
			}
```